# CS513 Data Cleaning Before and After Queries

Load SQL and Files

In [1]:
# Set-up
%load_ext sql
%sql sqlite://
import pandas as pd

In [2]:
!git clone https://github.com/hhlim2/cs513_team6.git

Cloning into 'cs513_team6'...
remote: Enumerating objects: 10, done.
remote: Counting objects: 100% (10/10), done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 10 (delta 4), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (10/10), 7.64 MiB | 13.30 MiB/s, done.
Resolving deltas: 100% (4/4), done.


In [3]:
!unzip /content/cs513_team6/Dish.zip

Archive:  /content/cs513_team6/Dish.zip
  inflating: Dish.csv                


In [4]:
path_to_original_file = 'Dish.csv'
path_to_cleaned_file = '/content/cs513_team6/Data-Cleaning-Dish_Final_15Jul.csv'

In [5]:
# Create database tables from CSV files

with open(path_to_original_file) as f: Dish = pd.read_csv(path_to_original_file, encoding='unicode_escape', index_col=0)
%sql drop table if exists Dish;
%sql --persist Dish

 * sqlite://
Done.
 * sqlite://


'Persisted dish'

In [6]:
with open(path_to_cleaned_file) as f: DishCleaned = pd.read_csv(path_to_cleaned_file, encoding='unicode_escape', index_col=0)
%sql drop table if exists DishCleaned;
%sql --persist DishCleaned

 * sqlite://
Done.
 * sqlite://


'Persisted dishcleaned'

## Data Cleaning Before and After

### Find all non lowercase values in name

In [7]:
%%sql
--this query is too large and cannot complete with the runtime constraints therefore we limit the results to 100
select * from Dish where name != lower(name) limit 100

 * sqlite://
Done.


[(1, 'Consomme printaniere royal', None, 8, 8, 1897, 1927, 0.2, 0.4),
 (2, 'Chicken gumbo', None, 111, 117, 1895, 1960, 0.1, 0.8),
 (3, 'Tomato aux croutons', None, 13, 13, 1893, 1917, 0.25, 0.4),
 (4, 'Onion au gratin', None, 41, 41, 1900, 1971, 0.25, 1.0),
 (5, 'St. Emilion', None, 66, 68, 1881, 1981, 0.0, 18.0),
 (7, 'Radishes', None, 3262, 3346, 1854, 2928, 0.0, 25.0),
 (8, 'Chicken soup with rice', None, 48, 49, 1897, 1961, 0.1, 0.6),
 (9, 'Clam broth (cup)', None, 14, 16, 1899, 1962, 0.15, 0.4),
 (10, 'Cream of new asparagus, croutons', None, 2, 2, 1900, 1900, 0.0, 0.0),
 (11, 'Clear green turtle', None, 157, 157, 1893, 1937, 0.25, 60.0),
 (12, 'Striped bass saute, meuniere', None, 2, 2, 1900, 1900, 0.0, 0.0),
 (13, 'Anchovies', None, 453, 484, 1858, 1987, 0.0, 30.0),
 (14, 'Fresh lobsters in every style', None, 4, 4, 1899, 1900, 0.0, 0.0),
 (15, 'Celery', None, 4246, 4690, 1, 2928, 0.0, 50.0),
 (16, 'Pim-olas', None, 145, 148, 1897, 1918, 0.15, 35.0),
 (17, 'Caviar', None, 505, 534, 1880, 1987, 0.0, 75.0),
 (18, 'Sardines', None, 1425, 1484, 1856, 2928, 0.0, 50.0),
 (19, 'India chutney', None, 16, 16, 1865, 1901, 0.1, 0.2),
 (20, 'Pickles', None, 453, 472, 1852, 1987, 0.0, 10.0),
 (21, 'English walnuts', None, 83, 86, 1851, 1948, 0.1, 0.3),
 (22, 'Pate de foies-gras', None, 9, 9, 1898, 1901, 1.0, 1.0),
 (23, 'Pomard', None, 11, 11, 1880, 1950, 0.75, 5.0),
 (24, 'Brook trout, mountain style', None, 2, 2, 1900, 1900, 0.0, 0.0),
 (25, 'Whitebait, sauce tartare', None, 4, 4, 1900, 1901, 0.3, 0.75),
 (26, 'Clams', None, 170, 180, 1881, 1970, 0.1, 0.9),
 (27, 'Oysters', None, 289, 292, 1862, 1963, 0.0, 35.0),
 (28, 'Claremont planked shad', None, 2, 2, 1899, 1900, 1.5, 2.5),
 (29, "G. H. Mumm & Co's Extra Dry", None, 14, 14, 1895, 1914, 2.0, 4.0),
 (30, 'Cerealine with Milk', None, 1, 1, 1901, 1901, 0.0, 0.0),
 (31, 'Sliced Bananas', None, 220, 238, 1900, 1987, 0.0, 15.0),
 (32, 'Wheat Vitos', None, 3, 3, 1900, 1900, 0.0, 0.0),
 (33, 'Sliced Tomatoes', None, 1195, 1256, 1873, 2928, 0.0, 25.0),
 (34, 'Russian Caviare on Toast', None, 3, 3, 1900, 1900, None, None),
 (35, 'Smoked beef in cream', None, 21, 21, 1896, 1933, 0.3, 0.9),
 (38, 'Apple Sauce', None, 721, 829, 1, 1987, 0.0, 20.0),
 (39, 'Potage a la Victoria', None, 5, 5, 1899, 1901, None, None),
 (40, 'Breakfast', None, 28, 28, 1900, 1967, 0.25, 1.0),
 (41, 'Strawberries', None, 304, 339, 1854, 1985, 0.0, 50.0),
 (42, 'Preserved figs', None, 257, 259, 1889, 1969, 0.0, 30.0),
 (43, 'BLUE POINTS', None, 868, 891, 1881, 1968, 0.15, 50.0),
 (44, 'CONSOMME ANGLAISE', None, 5, 5, 1900, 1907, 0.2, 0.25),
 (45, 'CREAM OF CAULIFLOWER', None, 18, 18, 1899, 1949, 0.1, 0.5),
 (46, "BROILED SHAD, A LA MAITRE D'HOTEL", None, 6, 6, 1896, 1900, 0.0, 0.0),
 (47, 'SLICED CUCUMBERS', None, 455, 458, 1881, 1964, 0.0, 25.0),
 (48, 'SALTED ALMONDS', None, 773, 782, 1889, 2012, 0.0, 10.0),
 (49, 'POTATOES, JULIEN', None, 3, 3, 1900, 1900, 0.0, 0.0),
 (50, 'Cracked Wheat', None, 120, 124, 1859, 1917, 0.1, 0.3),
 (51, 'Malt Breakfast Food', None, 63, 64, 1899, 1917, 0.1, 0.35),
 (52, 'BOILED BEEF TONGUE, ITALIAN SAUCE', None, 3, 3, 1900, 1900, 0.0, 0.0),
 (53, 'Young Onions', None, 157, 157, 1894, 1949, 0.05, 10.0),
 (54, 'Pears', None, 631, 650, 1852, 1974, 0.0, 50.0),
 (55, 'ROAST SIRLOIN OF BEEF, YORKSHIRE PUDDING', None, 11, 12, 1900, 1974, 0.0, 0.75),
 (57, 'Huhnerbruhe', None, 11, 11, 1900, 1901, 0.0, 0.0),
 (58, 'ROAST EASTER LAMB, MINT SAUCE', None, 4, 4, 1899, 1900, 0.0, 0.0),
 (59, 'Rockaways', None, 157, 162, 1889, 1948, 0.2, 50.0),
 (60, 'Hafergrutze', None, 205, 218, 1899, 1910, None, None),
 (61, 'BROWNED POTATOES', None, 55, 58, 1893, 1980, 0.25, 20.0),
 (62, 'Pampelmuse', None, 30, 30, 1900, 1910, 0.0, 0.0),
 (63, 'Apfelsinen', None, 181, 184, 1899, 1935, None, None),
 (64, 'Ananas', None, 94, 104, 1884, 1978, 0.0, 0.95),
 (65, 'Milchreis', None, 135, 135, 1899, 1910, None, None),
 (66, 'Grape fruit', None, 715, 749, 1895, 1968, 0.0, 2540.0),
 (67, 'Oranges', None, 1423, 1479, 1

In [8]:
%%sql

select * from DishCleaned where name_cluster_coffee != lower(name_cluster_coffee)

 * sqlite://
Done.


[]

### Check to make sure that all values in cleaned dataset are coffee related

In [9]:
#this query is too large and cannot complete with the runtime constraints therefore we limit the results to 100
%%sql
SELECT * FROM Dish
WHERE
    name NOT LIKE '%americano (coffee)%' AND
name NOT LIKE '%blue mountain coffee%' AND
name NOT LIKE '%calypso coffee%' AND
name NOT LIKE '%coffee%' AND
name NOT LIKE '%coffee (decaf)%' AND
name NOT LIKE '%coffee (large pot)%' AND
name NOT LIKE '%coffee (pot)%' AND
name NOT LIKE '%coffee (small pot)%' AND
name NOT LIKE '%coffee bavaroise%' AND
name NOT LIKE '%coffee biscuit%' AND
name NOT LIKE '%coffee blancmange%' AND
name NOT LIKE '%coffee cake%' AND
name NOT LIKE '%coffee cocktail%' AND
name NOT LIKE '%coffee drink%' AND
name NOT LIKE '%coffee eclair%' AND
name NOT LIKE '%coffee glacé%' AND
name NOT LIKE '%coffee granita%' AND
name NOT LIKE '%coffee ice cream%' AND
name NOT LIKE '%coffee in tin%' AND
name NOT LIKE '%coffee jelly%' AND
name NOT LIKE '%coffee liqueur%' AND
name NOT LIKE '%coffee mousse%' AND
name NOT LIKE '%coffee parfait%' AND
name NOT LIKE '%coffee pastry%' AND
name NOT LIKE '%coffee pudding%' AND
name NOT LIKE '%coffee sambuca%' AND
name NOT LIKE '%coffee souffle%' AND
name NOT LIKE '%coffee sundae%' AND
name NOT LIKE '%coffee with cigar%' AND
name NOT LIKE '%coffee with food%' AND
name NOT LIKE '%cuban coffee%' AND
name NOT LIKE '%dutch coffee%' AND
name NOT LIKE '%frappe (coffee)%' AND
name NOT LIKE '%french coffee%' AND
name NOT LIKE '%french coffee (decaf)%' AND
name NOT LIKE '%iced americano (coffee)%' AND
name NOT LIKE '%iced coffee%' AND
name NOT LIKE '%iced coffee (pot)%' AND
name NOT LIKE '%iced frappe (coffee)%' AND
name NOT LIKE '%irish coffee%' AND
name NOT LIKE '%kona coffee%' AND
name NOT LIKE "%l'absinthe coffee%" AND
name NOT LIKE '%latte (coffee)%' AND
name NOT LIKE '%macchiato (coffee)%' AND
name NOT LIKE '%mocha (coffee)%' AND
name NOT LIKE '%new orleans coffee%' AND
name NOT LIKE '%postum (coffee substitute)%' AND
name NOT LIKE '%russian coffee%' AND
name NOT LIKE '%spanish coffee%' AND
name NOT LIKE '%swiss coffee%' AND
name NOT LIKE '%toulouse lautrec coffee%' AND
name NOT LIKE '%turkish coffee%' AND
name NOT LIKE '%cafe au lait%' AND
name NOT LIKE '%cafe au lait (pot)%' AND
name NOT LIKE '%cafe brulot%' AND
name NOT LIKE '%cafe con leche%' AND
name NOT LIKE '%cafe liegeois%' AND
name NOT LIKE '%iced cafe au lait%' AND
name NOT LIKE '%espresso%' AND
name NOT LIKE '%espresso (decaf)%' AND
name NOT LIKE '%espresso (pot)%' AND
name NOT LIKE '%espresso cake%' AND
name NOT LIKE '%espresso cocktail%' AND
name NOT LIKE '%espresso drink%' AND
name NOT LIKE '%espresso pastry%' AND
name NOT LIKE '%espresso wafers%' AND
name NOT LIKE '%espresso with food%' AND
name NOT LIKE '%iced espresso%' AND
name NOT LIKE '%tiramisu (espresso)%' AND
name NOT LIKE '%iced mocha%' AND
name NOT LIKE '%mocha (pot)%' AND
name NOT LIKE '%mocha bavaroise%' AND
name NOT LIKE '%mocha cake%' AND
name NOT LIKE '%mocha drink%' AND
name NOT LIKE '%mocha eclair%' AND
name NOT LIKE '%mocha glacé%' AND
name NOT LIKE '%mocha ice cream%' AND
name NOT LIKE '%mocha jelly%' AND
name NOT LIKE '%mocha macaron%' AND
name NOT LIKE '%mocha mousse%' AND
name NOT LIKE '%mocha parfait%' AND
name NOT LIKE '%mocha pastry%' AND
name NOT LIKE '%mocha pie%' AND
name NOT LIKE '%mocha praline%' AND
name NOT LIKE '%mocha pudding%' AND
name NOT LIKE '%mocha souffle%' AND
name NOT LIKE '%mocha sundae%' AND
name NOT LIKE '%mocha tart%' AND
name NOT LIKE '%mocha torte%' AND
name NOT LIKE '%mocha with food%' AND
name NOT LIKE '%mochaccino%' AND
name NOT LIKE '%cappuccino%' AND
name NOT LIKE '%cappuccino (decaf)%' AND
name NOT LIKE '%cappuccino cocktail%' AND
name NOT LIKE '%cappuccino with food%' AND
name NOT LIKE '%iced cappuccino%' AND
name NOT LIKE '%ristretto%' AND
name NOT LIKE '%ristretto cocktail%'
limit 100


 * sqlite://
Done.


[(1, 'Consomme printaniere royal', None, 8, 8, 1897, 1927, 0.2, 0.4),
 (2, 'Chicken gumbo', None, 111, 117, 1895, 1960, 0.1, 0.8),
 (3, 'Tomato aux croutons', None, 13, 13, 1893, 1917, 0.25, 0.4),
 (4, 'Onion au gratin', None, 41, 41, 1900, 1971, 0.25, 1.0),
 (5, 'St. Emilion', None, 66, 68, 1881, 1981, 0.0, 18.0),
 (7, 'Radishes', None, 3262, 3346, 1854, 2928, 0.0, 25.0),
 (8, 'Chicken soup with rice', None, 48, 49, 1897, 1961, 0.1, 0.6),
 (9, 'Clam broth (cup)', None, 14, 16, 1899, 1962, 0.15, 0.4),
 (10, 'Cream of new asparagus, croutons', None, 2, 2, 1900, 1900, 0.0, 0.0),
 (11, 'Clear green turtle', None, 157, 157, 1893, 1937, 0.25, 60.0),
 (12, 'Striped bass saute, meuniere', None, 2, 2, 1900, 1900, 0.0, 0.0),
 (13, 'Anchovies', None, 453, 484, 1858, 1987, 0.0, 30.0),
 (14, 'Fresh lobsters in every style', None, 4, 4, 1899, 1900, 0.0, 0.0),
 (15, 'Celery', None, 4246, 4690, 1, 2928, 0.0, 50.0),
 (16, 'Pim-olas', None, 145, 148, 1897, 1918, 0.15, 35.0),
 (17, 'Caviar', None, 505, 534, 1880, 1987, 0.0, 75.0),
 (18, 'Sardines', None, 1425, 1484, 1856, 2928, 0.0, 50.0),
 (19, 'India chutney', None, 16, 16, 1865, 1901, 0.1, 0.2),
 (20, 'Pickles', None, 453, 472, 1852, 1987, 0.0, 10.0),
 (21, 'English walnuts', None, 83, 86, 1851, 1948, 0.1, 0.3),
 (22, 'Pate de foies-gras', None, 9, 9, 1898, 1901, 1.0, 1.0),
 (23, 'Pomard', None, 11, 11, 1880, 1950, 0.75, 5.0),
 (24, 'Brook trout, mountain style', None, 2, 2, 1900, 1900, 0.0, 0.0),
 (25, 'Whitebait, sauce tartare', None, 4, 4, 1900, 1901, 0.3, 0.75),
 (26, 'Clams', None, 170, 180, 1881, 1970, 0.1, 0.9),
 (27, 'Oysters', None, 289, 292, 1862, 1963, 0.0, 35.0),
 (28, 'Claremont planked shad', None, 2, 2, 1899, 1900, 1.5, 2.5),
 (29, "G. H. Mumm & Co's Extra Dry", None, 14, 14, 1895, 1914, 2.0, 4.0),
 (30, 'Cerealine with Milk', None, 1, 1, 1901, 1901, 0.0, 0.0),
 (31, 'Sliced Bananas', None, 220, 238, 1900, 1987, 0.0, 15.0),
 (32, 'Wheat Vitos', None, 3, 3, 1900, 1900, 0.0, 0.0),
 (33, 'Sliced Tomatoes', None, 1195, 1256, 1873, 2928, 0.0, 25.0),
 (34, 'Russian Caviare on Toast', None, 3, 3, 1900, 1900, None, None),
 (35, 'Smoked beef in cream', None, 21, 21, 1896, 1933, 0.3, 0.9),
 (38, 'Apple Sauce', None, 721, 829, 1, 1987, 0.0, 20.0),
 (39, 'Potage a la Victoria', None, 5, 5, 1899, 1901, None, None),
 (40, 'Breakfast', None, 28, 28, 1900, 1967, 0.25, 1.0),
 (41, 'Strawberries', None, 304, 339, 1854, 1985, 0.0, 50.0),
 (42, 'Preserved figs', None, 257, 259, 1889, 1969, 0.0, 30.0),
 (43, 'BLUE POINTS', None, 868, 891, 1881, 1968, 0.15, 50.0),
 (44, 'CONSOMME ANGLAISE', None, 5, 5, 1900, 1907, 0.2, 0.25),
 (45, 'CREAM OF CAULIFLOWER', None, 18, 18, 1899, 1949, 0.1, 0.5),
 (46, "BROILED SHAD, A LA MAITRE D'HOTEL", None, 6, 6, 1896, 1900, 0.0, 0.0),
 (47, 'SLICED CUCUMBERS', None, 455, 458, 1881, 1964, 0.0, 25.0),
 (48, 'SALTED ALMONDS', None, 773, 782, 1889, 2012, 0.0, 10.0),
 (49, 'POTATOES, JULIEN', None, 3, 3, 1900, 1900, 0.0, 0.0),
 (50, 'Cracked Wheat', None, 120, 124, 1859, 1917, 0.1, 0.3),
 (51, 'Malt Breakfast Food', None, 63, 64, 1899, 1917, 0.1, 0.35),
 (52, 'BOILED BEEF TONGUE, ITALIAN SAUCE', None, 3, 3, 1900, 1900, 0.0, 0.0),
 (53, 'Young Onions', None, 157, 157, 1894, 1949, 0.05, 10.0),
 (54, 'Pears', None, 631, 650, 1852, 1974, 0.0, 50.0),
 (55, 'ROAST SIRLOIN OF BEEF, YORKSHIRE PUDDING', None, 11, 12, 1900, 1974, 0.0, 0.75),
 (57, 'Huhnerbruhe', None, 11, 11, 1900, 1901, 0.0, 0.0),
 (58, 'ROAST EASTER LAMB, MINT SAUCE', None, 4, 4, 1899, 1900, 0.0, 0.0),
 (59, 'Rockaways', None, 157, 162, 1889, 1948, 0.2, 50.0),
 (60, 'Hafergrutze', None, 205, 218, 1899, 1910, None, None),
 (61, 'BROWNED POTATOES', None, 55, 58, 1893, 1980, 0.25, 20.0),
 (62, 'Pampelmuse', None, 30, 30, 1900, 1910, 0.0, 0.0),
 (63, 'Apfelsinen', None, 181, 184, 1899, 1935, None, None),
 (64, 'Ananas', None, 94, 104, 1884, 1978, 0.0, 0.95),
 (65, 'Milchreis', None, 135, 135, 1899, 1910, None, None),
 (66, 'Grape fruit', None, 715, 749, 1895, 1968, 0.0, 2540.0),
 (67, 'Oranges', None, 1423, 1479, 1

In [10]:
%%sql

SELECT * FROM DishCleaned
WHERE
name_cluster_coffee NOT LIKE '%americano (coffee)%' AND
name_cluster_coffee NOT LIKE '%blue mountain coffee%' AND
name_cluster_coffee NOT LIKE '%calypso coffee%' AND
name_cluster_coffee NOT LIKE '%coffee%' AND
name_cluster_coffee NOT LIKE '%coffee (decaf)%' AND
name_cluster_coffee NOT LIKE '%coffee (large pot)%' AND
name_cluster_coffee NOT LIKE '%coffee (pot)%' AND
name_cluster_coffee NOT LIKE '%coffee (small pot)%' AND
name_cluster_coffee NOT LIKE '%coffee bavaroise%' AND
name_cluster_coffee NOT LIKE '%coffee biscuit%' AND
name_cluster_coffee NOT LIKE '%coffee blancmange%' AND
name_cluster_coffee NOT LIKE '%coffee cake%' AND
name_cluster_coffee NOT LIKE '%coffee cocktail%' AND
name_cluster_coffee NOT LIKE '%coffee drink%' AND
name_cluster_coffee NOT LIKE '%coffee eclair%' AND
name_cluster_coffee NOT LIKE '%coffee glacé%' AND
name_cluster_coffee NOT LIKE '%coffee granita%' AND
name_cluster_coffee NOT LIKE '%coffee ice cream%' AND
name_cluster_coffee NOT LIKE '%coffee in tin%' AND
name_cluster_coffee NOT LIKE '%coffee jelly%' AND
name_cluster_coffee NOT LIKE '%coffee liqueur%' AND
name_cluster_coffee NOT LIKE '%coffee mousse%' AND
name_cluster_coffee NOT LIKE '%coffee parfait%' AND
name_cluster_coffee NOT LIKE '%coffee pastry%' AND
name_cluster_coffee NOT LIKE '%coffee pudding%' AND
name_cluster_coffee NOT LIKE '%coffee sambuca%' AND
name_cluster_coffee NOT LIKE '%coffee souffle%' AND
name_cluster_coffee NOT LIKE '%coffee sundae%' AND
name_cluster_coffee NOT LIKE '%coffee with cigar%' AND
name_cluster_coffee NOT LIKE '%coffee with food%' AND
name_cluster_coffee NOT LIKE '%cuban coffee%' AND
name_cluster_coffee NOT LIKE '%dutch coffee%' AND
name_cluster_coffee NOT LIKE '%frappe (coffee)%' AND
name_cluster_coffee NOT LIKE '%french coffee%' AND
name_cluster_coffee NOT LIKE '%french coffee (decaf)%' AND
name_cluster_coffee NOT LIKE '%iced americano (coffee)%' AND
name_cluster_coffee NOT LIKE '%iced coffee%' AND
name_cluster_coffee NOT LIKE '%iced coffee (pot)%' AND
name_cluster_coffee NOT LIKE '%iced frappe (coffee)%' AND
name_cluster_coffee NOT LIKE '%irish coffee%' AND
name_cluster_coffee NOT LIKE '%kona coffee%' AND
name_cluster_coffee NOT LIKE "%l'absinthe coffee%" AND
name_cluster_coffee NOT LIKE '%latte (coffee)%' AND
name_cluster_coffee NOT LIKE '%macchiato (coffee)%' AND
name_cluster_coffee NOT LIKE '%mocha (coffee)%' AND
name_cluster_coffee NOT LIKE '%new orleans coffee%' AND
name_cluster_coffee NOT LIKE '%postum (coffee substitute)%' AND
name_cluster_coffee NOT LIKE '%russian coffee%' AND
name_cluster_coffee NOT LIKE '%spanish coffee%' AND
name_cluster_coffee NOT LIKE '%swiss coffee%' AND
name_cluster_coffee NOT LIKE '%toulouse lautrec coffee%' AND
name_cluster_coffee NOT LIKE '%turkish coffee%' AND
name_cluster_coffee NOT LIKE '%cafe au lait%' AND
name_cluster_coffee NOT LIKE '%cafe au lait (pot)%' AND
name_cluster_coffee NOT LIKE '%cafe brulot%' AND
name_cluster_coffee NOT LIKE '%cafe con leche%' AND
name_cluster_coffee NOT LIKE '%cafe liegeois%' AND
name_cluster_coffee NOT LIKE '%iced cafe au lait%' AND
name_cluster_coffee NOT LIKE '%espresso%' AND
name_cluster_coffee NOT LIKE '%espresso (decaf)%' AND
name_cluster_coffee NOT LIKE '%espresso (pot)%' AND
name_cluster_coffee NOT LIKE '%espresso cake%' AND
name_cluster_coffee NOT LIKE '%espresso cocktail%' AND
name_cluster_coffee NOT LIKE '%espresso drink%' AND
name_cluster_coffee NOT LIKE '%espresso pastry%' AND
name_cluster_coffee NOT LIKE '%espresso wafers%' AND
name_cluster_coffee NOT LIKE '%espresso with food%' AND
name_cluster_coffee NOT LIKE '%iced espresso%' AND
name_cluster_coffee NOT LIKE '%tiramisu (espresso)%' AND
name_cluster_coffee NOT LIKE '%iced mocha%' AND
name_cluster_coffee NOT LIKE '%mocha%' AND
name_cluster_coffee NOT LIKE '%mocha (pot)%' AND
name_cluster_coffee NOT LIKE '%mocha bavaroise%' AND
name_cluster_coffee NOT LIKE '%mocha cake%' AND
name_cluster_coffee NOT LIKE '%mocha drink%' AND
name_cluster_coffee NOT LIKE '%mocha eclair%' AND
name_cluster_coffee NOT LIKE '%mocha glacé%' AND
name_cluster_coffee NOT LIKE '%mocha ice cream%' AND
name_cluster_coffee NOT LIKE '%mocha jelly%' AND
name_cluster_coffee NOT LIKE '%mocha macaron%' AND
name_cluster_coffee NOT LIKE '%mocha mousse%' AND
name_cluster_coffee NOT LIKE '%mocha parfait%' AND
name_cluster_coffee NOT LIKE '%mocha pastry%' AND
name_cluster_coffee NOT LIKE '%mocha pie%' AND
name_cluster_coffee NOT LIKE '%mocha praline%' AND
name_cluster_coffee NOT LIKE '%mocha pudding%' AND
name_cluster_coffee NOT LIKE '%mocha souffle%' AND
name_cluster_coffee NOT LIKE '%mocha sundae%' AND
name_cluster_coffee NOT LIKE '%mocha tart%' AND
name_cluster_coffee NOT LIKE '%mocha torte%' AND
name_cluster_coffee NOT LIKE '%mocha with food%' AND
name_cluster_coffee NOT LIKE '%mochaccino%' AND
name_cluster_coffee NOT LIKE '%cappuccino%' AND
name_cluster_coffee NOT LIKE '%cappuccino (decaf)%' AND
name_cluster_coffee NOT LIKE '%cappuccino cocktail%' AND
name_cluster_coffee NOT LIKE '%cappuccino with food%' AND
name_cluster_coffee NOT LIKE '%iced cappuccino%' AND
name_cluster_coffee NOT LIKE '%ristretto%' AND
name_cluster_coffee NOT LIKE '%ristretto cocktail%'



 * sqlite://
Done.


[]

### Check for values where menus_appeared < 1

In [11]:
%%sql
select * from Dish where menus_appeared < 1 limit 100

 * sqlite://
Done.


[(88446, 'toddy', None, 0, 1, 0, 1969, 0.0, 0.0),
 (132992, 'Pate en Croute', None, 0, 1, 0, 0, 0.0, 0.0),
 (164029, 'Clear beef broth', None, 0, 1, 1900, 0, 0.25, 0.25),
 (187708, 'kalbssteak mit champignons', None, 0, 1, 0, 0, 0.0, 0.0),
 (204888, 'Hot roast beef with gravy', None, 0, 1, 1900, 0, 0.25, 0.25),
 (212649, 'LE TOURNEDOS au poivre vert', None, 0, 1, 0, 1985, 26.0, 26.0),
 (218524, 'SOFT SHELL CRABS, SAUTE AMANDINE', None, 0, 1, 0, 1977, 7.25, 7.25),
 (241581, 'Turtle fin Baltimore', None, 0, 1, 0, 0, 5.75, 5.75),
 (243807, 'Soup Of The Day (Bowl)', None, 0, 1, 0, 0, 1.95, 1.95),
 (250693, "SURI LEBERLI - Shredded Calf's Liver Flambe in White Wine, Roesti", None, 0, 1, 1945, 0, None, None),
 (250699, 'SWISS MINCED VEAL, ROESTI', None, 0, 1, 1945, 0, None, None),
 (289679, 'Beetroot Salad a la Russe', None, 0, 1, 0, 0, 0.0, 0.0),
 (289680, 'Cold Eggs Fantasio', None, 0, 1, 0, 0, 0.0, 0.0),
 (289682, 'Bologna  Mortadella', None, 0, 1, 0, 0, 0.0, 0.0),
 (289683, 'Vicomte Salad', None, 0, 1, 0, 0, 0.0, 0.0),
 (289684, 'Terrine of Minced Pork', None, 0, 1, 0, 0, 0.0, 0.0),
 (289686, 'Omelet Basquaise', None, 0, 1, 0, 0, 0.0, 0.0),
 (289688, 'Supreme of Pampano with Almonds', None, 0, 1, 0, 0, 0.0, 0.0),
 (289689, "Braised Prague Ham a l'Auxoise", None, 0, 1, 0, 0, 0.0, 0.0),
 (289690, 'Escalope of Veal Poelee "Liberte"', None, 0, 1, 0, 0, 0.0, 0.0),
 (289696, 'Gnocchi a la Parisienne', None, 0, 1, 0, 0, 0.0, 0.0),
 (289697, 'Broiled Prime Sirloin Steak Bearnaise', None, 0, 1, 0, 0, 0.0, 0.0),
 (289698, 'Virginia Ham with Spices', None, 0, 1, 0, 0, 0.0, 0.0),
 (289699, 'Slice of Duckling with Cherries', None, 0, 1, 0, 0, 0.0, 0.0),
 (289701, 'Romaine Salad with Chopped Herbs', None, 0, 1, 0, 0, 0.0, 0.0),
 (289703, 'Scarole Salad with Cucumber', None, 0, 1, 0, 0, 0.0, 0.0),
 (289708, 'Chocolate Ice Cream a la Francaise', None, 0, 1, 0, 0, 0.0, 0.0),
 (289709, 'Meringuette with Almonds', None, 0, 1, 0, 0, 0.0, 0.0),
 (301736, 'Cafe Glacee', None, 0, 2, 1940, 0, 0.4, 0.4),
 (304667, 'Chilean Red Wine', None, 0, 1, 0, 1945, 1.5, 3.0),
 (306637, 'CARRE DE PORC FROID', None, 0, 1, 0, 1944, 0.0, 0.0),
 (307821, 'Deviled Smithfield Ham Sandwich with a Sliced Tomato', None, 0, 1, 0, 0, 0.45, 0.45),
 (307822, 'Lackawanna Club Sandwich', None, 0, 1, 0, 0, 0.75, 0.75),
 (307832, 'El Dorado Imported Rum', None, 0, 1, 0, 0, 0.45, 0.45),
 (307833, "Gilbey's London Dry Gin", None, 0, 1, 0, 0, 0.45, 0.45),
 (307835, 'Jameson Dublin Irish Whiskey', None, 0, 1, 0, 0, 0.6, 0.6),
 (307837, 'Lejon Three-Star Brandy', None, 0, 1, 0, 0, 0.44, 0.44),
 (307838, "Jacquin's Liqueur de St. Dominic (Benedictine Type)", None, 0, 1, 0, 0, 0.45, 0.45),
 (307840, 'Domestic Beer, Popular Brands', None, 0, 1, 0, 0, 0.25, 0.25),
 (307842, "Ruppert's Special Brew Ale", None, 0, 1, 0, 0, 0.3, 0.3),
 (307845, 'Champagne, Gold Seal, Native, Half Bottle', None, 0, 1, 0, 0, 2.15, 2.15),
 (307846, 'Burgundy, Beaulieu Vineyard, Native, Half Bottle', None, 0, 1, 0, 0, 1.11, 1.11),
 (307848, 'Riesling, Beaulieu Vineyard, Native, Half Bottle', None, 0, 1, 0, 0, 1.11, 1.11),
 (307849, 'Sauterne, Great Western, Native, Half Bottle', None, 0, 1, 0, 0, 1.0, 1.0),
 (307855, 'Port, Choice Tawny, Cockburn, Imported, Ind. Bottle', None, 0, 1, 0, 0, 0.36, 0.36),
 (307857, 'Sherry, Santa Maria, Duff Gordon, Imported, Ind. Bottle', None, 0, 1, 0, 0, 0.36, 0.36),
 (307858, 'Portland Water', None, 0, 1, 0, 0, 0.25, 0.25),
 (307860, 'Pluto Aperient', None, 0, 1, 0, 0, 0.15, 0.15),
 (308162, 'Chicne Salad Special on Toast, Sliced Tomato', None, 0, 1, 0, 0, 0.0, 0.0),
 (309262, 'Sliced Long Island Duckling', None, 0, 3, 0, 1947, 2.1, 2.1),
 (309629, 'Garlic Butter', None, 0, 1, 1947, 0, 0.4, 0.4),
 (310119, 'Freshly Squeezed Oranges', None, 0, 1, 0, 0, 2.25, 2.25),
 (310130, 'SEELBACH CONTINENTAL: Choice of Juice, Choice of Brioche, Croissant or Danish Pastry, Choice of Breakfast Bread, Muffin or Toast, Choice of Coffee, tea, Hot-Chocolate, Milk or Decaffeinated Coffee', None, 0, 1

In [12]:
%%sql
select * from DishCleaned where menus_appeared < 1

 * sqlite://
Done.


[]

### Check for missing values in lowest_price or highest_price

In [13]:
%%sql
select * from Dish where lowest_price is NULL or highest_price is NULL limit 100

 * sqlite://
Done.


[(34, 'Russian Caviare on Toast', None, 3, 3, 1900, 1900, None, None),
 (39, 'Potage a la Victoria', None, 5, 5, 1899, 1901, None, None),
 (60, 'Hafergrutze', None, 205, 218, 1899, 1910, None, None),
 (63, 'Apfelsinen', None, 181, 184, 1899, 1935, None, None),
 (65, 'Milchreis', None, 135, 135, 1899, 1910, None, None),
 (87, 'Hot or cold ribs of beef', None, 1, 1, 1900, 1900, None, None),
 (135, "Consomme aux Quenelle's", None, 2, 2, 1900, 1900, None, None),
 (136, 'Milk rice', None, 44, 48, 1900, 1970, None, None),
 (170, 'Baked Stuffed Mullet & Sauce Pomard', None, 2, 2, 1900, 1900, None, None),
 (346, 'Grilled Mutton Chops', None, 35, 35, 1892, 1910, None, None),
 (385, 'Eggs to order', None, 263, 264, 1891, 1973, None, None),
 (427, 'Sirloin or Tenderloin Steak', None, 3, 3, 1900, 1901, None, None),
 (433, 'Eggs as Ordered', None, 82, 83, 1898, 1949, None, None),
 (507, 'Fried & Boiled Potatoes', None, 8, 8, 1899, 1901, None, None),
 (527, 'Roast (Vancouver) Suckling Pig & Bread Sauce', None, 2, 2, 1900, 1900, None, None),
 (534, 'Fillet de Boeuf a la Parisienne', None, 2, 2, 1900, 1900, None, None),
 (538, 'Bouchees aux Macaroni', None, 2, 2, 1900, 1900, None, None),
 (543, 'Roast Goose & Apple Sauce', None, 3, 3, 1900, 1900, None, None),
 (555, 'Mashed Pumpkins', None, 3, 3, 1900, 1900, None, None),
 (558, 'Indo Curry & Rice', None, 2, 2, 1900, 1900, None, None),
 (559, 'Steamed Apple Pudding & Hard Sauce', None, 2, 2, 1900, 1900, None, None),
 (564, 'Scotch Sandwiches', None, 2, 2, 1900, 1900, None, None),
 (565, 'Chocolate Cream & Sauce', None, 2, 2, 1900, 1900, None, None),
 (567, 'Swiss Rolls', None, 7, 7, 1900, 1900, None, None),
 (571, 'Jam and Marmalade', None, 10, 10, 1900, 1910, None, None),
 (573, 'Cum-quats', None, 3, 3, 1900, 1900, None, None),
 (576, 'Fillet Steak and Lamb Chops to Order', None, 1, 1, 1901, 1901, None, None),
 (583, 'MacLaren Cheese', None, 16, 16, 1896, 1900, None, None),
 (584, 'China & Indian Teas', None, 9, 9, 1900, 1900, None, None),
 (587, 'Eggs Bacon and Ham to Order', None, 2, 2, 1901, 1901, None, None),
 (596, 'Tea Coffee and Cocoa', None, 2, 2, 1901, 1901, None, None),
 (609, 'Cotelettes de Veau, Sauce Tomate', None, 3, 3, 1901, 1901, None, None),
 (627, 'A. A. Oats with Fresh Milk', None, 1, 1, 1901, 1901, None, None),
 (642, 'Eggs, Bacon and Ham to Order', None, 2, 2, 1901, 1901, None, None),
 (648, 'Egg to Order', None, 17, 17, 1900, 1948, None, None),
 (654, 'Jam & Marmalade', None, 3, 3, 1900, 1900, None, None),
 (656, 'Fruit.', None, 37, 38, 1889, 1901, None, None),
 (740, 'Fried and broiled Ham', None, 4, 4, 1901, 1901, None, None),
 (744, 'Potatoes boiled and Mashed', None, 8, 8, 1900, 1901, None, None),
 (771, 'Cliquot Sec', None, 1, 1, 1889, 1891, None, None),
 (793, "Broiled Shad Maitre d'Hotel", None, 3, 3, 1900, 1913, None, None),
 (809, 'Obst.', None, 12, 12, 1899, 1901, None, None),
 (811, 'Nachtisch.', None, 4, 4, 1900, 1901, None, None),
 (814, 'Batter Cakes', None, 8, 8, 1900, 1901, None, None),
 (818, 'Pork Cutlets', None, 26, 30, 1901, 1910, None, None),
 (822, 'Tea & Coffee', None, 11, 11, 1900, 1901, None, None),
 (835, 'Cornbread', None, 24, 24, 1900, 1901, None, None),
 (842, 'Semolina Pudding', None, 17, 17, 1900, 1938, None, None),
 (855, 'Coffee and Tea', None, 11, 11, 1900, 1976, None, None),
 (887, 'Broiled Mutton Chops', None, 52, 52, 1859, 1910, None, None),
 (912, 'Bratapfel mit Schlagsahne', None, 13, 13, 1899, 1900, None, None),
 (925, 'Salzmakrele', None, 57, 57, 1899, 1907, None, None),
 (928, 'Gekochter Reis', None, 64, 64, 1900, 1935, None, None),
 (932, 'Delikatess Haringe', None, 35, 35, 1899, 1901, None, None),
 (935, 'Mutton Chops & Chips', None, 9, 9, 1900, 1906, None, None),
 (937, 'Sulze', None, 91, 91, 1899, 1907, None, None),
 (940, 'Gekochter Yorkshire Schinken', None, 62, 63, 1899, 1910, None, None),
 (1029, 'Entrees', None, 67, 70, 1897, 2006, None, None),
 (1035, 'Obst', None, 177, 177, 1888, 1938, None, None),
 (1036, 'Nachtisch',

In [14]:
%%sql

select * from DishCleaned where lowest_price is NULL or highest_price is NULL

 * sqlite://
Done.


[]

### Check for invalid dates

In [15]:
%%sql

select * from Dish where first_appeared < 1840  or last_appeared > 2021 limit 100

 * sqlite://
Done.


[(7, 'Radishes', None, 3262, 3346, 1854, 2928, 0.0, 25.0),
 (15, 'Celery', None, 4246, 4690, 1, 2928, 0.0, 50.0),
 (18, 'Sardines', None, 1425, 1484, 1856, 2928, 0.0, 50.0),
 (33, 'Sliced Tomatoes', None, 1195, 1256, 1873, 2928, 0.0, 25.0),
 (38, 'Apple Sauce', None, 721, 829, 1, 1987, 0.0, 20.0),
 (96, 'Coffee', None, 7740, 8484, 1, 2928, 0.0, 30.0),
 (103, 'Kippered Herring', None, 231, 236, 1, 1970, 0.0, 0.85),
 (108, 'Bananas', None, 1195, 1213, 1, 2012, 0.0, 25.0),
 (112, 'Fruit', None, 1919, 2006, 1854, 2928, 0.0, 40.0),
 (155, 'Oatmeal', None, 532, 610, 1, 1988, 0.0, 20.0),
 (169, 'Club sandwich', None, 639, 657, 1896, 2928, 0.0, 40.0),
 (239, 'Broiled Ham', None, 622, 635, 1, 1970, 0.0, 25.0),
 (247, 'Sirloin Steak', None, 912, 929, 1, 1987, 0.0, 235.0),
 (265, 'Stewed', None, 49, 55, 1, 1918, 0.1, 15.0),
 (270, 'Hominy', None, 421, 510, 1, 1969, 0.1, 0.35),
 (280, 'Corn Cakes', None, 146, 148, 1, 1953, 0.02, 0.55),
 (293, 'Lyonnaise Potatoes', None, 728, 746, 1, 1989, 0.0, 30.0),
 (340, 'Raw on the Half Shell', None, 1, 1, 0, 0, 0.0, 0.0),
 (348, 'Smelts', None, 93, 95, 1, 1998, 0.15, 50.0),
 (352, 'Buckwheat Cakes', None, 346, 350, 1, 1962, 0.0, 20.0),
 (370, 'Corned Beef Hash', None, 616, 675, 1, 1959, 0.1, 25.0),
 (420, 'Codfish in Cream', None, 89, 90, 1, 1941, 0.0, 0.6),
 (423, "Calf's Liver and Bacon", None, 362, 374, 1, 1967, 0.0, 40.0),
 (441, 'Stewed in Cream', None, 26, 27, 1900, 2928, 0.15, 40.0),
 (511, 'English Muffins', None, 130, 136, 1, 1987, 0.0, 0.3),
 (513, 'Rice Cakes', None, 218, 224, 1, 1949, 0.05, 20.0),
 (569, 'Cup Custard', None, 885, 953, 1854, 2928, 0.0, 25.0),
 (575, 'Orange', None, 862, 972, 1, 1975, 0.0, 25.0),
 (644, 'Wheat Cakes', None, 200, 207, 1, 1959, 0.0, 15.0),
 (663, 'Curried Chicken with Rice', None, 152, 157, 1900, 2928, 0.0, 3.0),
 (691, 'Lamb Chops', None, 614, 624, 1, 1988, 0.0, 75.0),
 (725, 'Lynnhavens', None, 354, 423, 1896, 2928, 0.0, 50.0),
 (737, 'Potatoes', None, 302, 338, 1856, 2928, 0.0, 15.0),
 (739, 'Cherrystones', None, 519, 524, 1891, 2928, 0.0, 45.0),
 (776, 'Apollinaris', None, 1366, 1474, 1, 1988, 0.0, 40.25),
 (777, 'Johannis', None, 384, 479, 1, 1918, 0.0, 40.25),
 (795, 'Cheese', None, 1558, 1595, 1, 1990, 0.0, 40.0),
 (838, 'Cucumber', None, 326, 360, 1887, 2928, 0.0, 50.0),
 (873, 'String Beans', None, 1488, 1570, 1858, 2928, 0.0, 60.0),
 (911, 'Cotuits', None, 300, 345, 1900, 2928, 0.0, 45.0),
 (926, 'Beefsteak', None, 230, 306, 1, 1958, 0.0, 15.0),
 (1000, 'Pork Chop', None, 71, 72, 1, 1974, 0.0, 30.0),
 (1052, 'Dessert', None, 773, 819, 1864, 2928, 0.0, 40.0),
 (1099, 'Ham', None, 1249, 1336, 1851, 2928, 0.0, 95.0),
 (1103, 'Saute Potatoes', None, 383, 395, 1, 1973, 0.0, 20.0),
 (1105, 'Watercress', None, 360, 365, 1891, 2928, 0.1, 30.0),
 (1202, 'Soft Shell Crabs', None, 177, 187, 1, 2006, 0.15, 60.0),
 (1206, 'Halibut', None, 122, 127, 1, 1965, 0.0, 60.0),
 (1557, 'Lobster', None, 306, 341, 1856, 2928, 0.0, 75.0),
 (1563, 'Dandelion', None, 37, 38, 1892, 2928, 0.15, 0.6),
 (1587, 'Spaghetti', None, 245, 299, 1895, 2928, 0.0, 50.0),
 (1653, 'Smoked Salmon', None, 590, 637, 1, 1993, 0.0, 50.0),
 (1822, 'Halibut steak', None, 196, 198, 1889, 2928, 0.0, 12.25),
 (1846, 'Fried Egg Plant', None, 662, 708, 1852, 2928, 0.0, 30.0),
 (1867, 'Shrimp', None, 155, 160, 1891, 2928, 0.0, 60.0),
 (1878, 'Little Necks', None, 676, 705, 1895, 2928, 0.0, 40.0),
 (1960, 'Fresh Mackerel', None, 160, 164, 1, 1940, 0.15, 60.0),
 (2002, 'Tenderloin', None, 112, 114, 1, 1971, 0.0, 100.0),
 (2005, 'cheese or ham', None, 1, 1, 0, 0, 0.0, 0.0),
 (2055, 'Waldorf', None, 79, 84, 1895, 2928, 0.0, 60.0),
 (2056, '40', None, 4, 4, 0, 0, 0.0, 0.0),
 (2057, 'Chiffonade', None, 128, 135, 1896, 2928, 0.0, 75.0),
 (2133, 'Weakfish', None, 51, 52, 1, 1917, 0.2, 30.5),
 (2265, 'Brandied Figs', None, 39, 39, 1, 1917, 0.25, 0.5),
 (2285, 'Macaroons', None, 654, 666, 1851, 2928, 0.0, 25.0),
 (2430, 'Currant Jelly', None, 334, 341, 1, 1965, 0.05, 20.0),
 (2572, 'Neapolitan', None, 168, 170, 1

In [16]:
%%sql
select * from DishCleaned where first_appeared_fixed < 1840 or last_appeared_fixed > 2021

 * sqlite://
Done.


[]

In [17]:
%%sql
select * from Dish where first_appeared > last_appeared

 * sqlite://
Done.


[(164029, 'Clear beef broth', None, 0, 1, 1900, 0, 0.25, 0.25),
 (204888, 'Hot roast beef with gravy', None, 0, 1, 1900, 0, 0.25, 0.25),
 (250693, "SURI LEBERLI - Shredded Calf's Liver Flambe in White Wine, Roesti", None, 0, 1, 1945, 0, None, None),
 (250699, 'SWISS MINCED VEAL, ROESTI', None, 0, 1, 1945, 0, None, None),
 (301736, 'Cafe Glacee', None, 0, 2, 1940, 0, 0.4, 0.4),
 (309629, 'Garlic Butter', None, 0, 1, 1947, 0, 0.4, 0.4)]

In [18]:
%%sql

select * from DishCleaned where first_appeared_fixed > last_appeared_fixed

 * sqlite://
Done.


[]

## UC1 Related Queries Using the Cleaned Dataset

### Q1: When did different types of coffee drinks (dishes) begin appearing on menus and become more prevalent?

In [ ]:
%%sql
select name, first_appeared_fixed, times_appeared from DishCleaned group by first_appeared_fixed

 * sqlite://
Done.


[('Coffee Royal', 1856, 3),
 ('Mocha Coffee', 1858, 8),
 ('Hot Coffee', 1862, 37),
 ('CAFE', 1866, 885),
 ('French Coffee', 1881, 517),
 ('coffee, per cup', 1882, 124),
 ('Cafe noir', 1883, 591),
 ('Cup Coffee with 2 Corn Muffins', 1884, 2),
 ('Coffee Cream', 1885, 3),
 ('Coffi', 1886, 4),
 ('Cafe Parfait', 1887, 647),
 ('Kaffee', 1888, 868),
 ('Caffe', 1889, 46),
 ('Coffee de turin', 1890, 1),
 ('Cafe au Lait', 1891, 21),
 ('Demi-tasse', 1892, 878),
 ('Coffee Cake', 1893, 135),
 ('Iced coffee', 1894, 602),
 ('Turkish Coffee', 1895, 623),
 ('Mousse au cafe a la Netherland', 1896, 6),
 ('Mocha', 1897, 57),
 ('Coffee, Pot', 1898, 173),
 ('Kaffee,', 1899, 5),
 ('Coffee', 1900, 8484),
 ('Coffee, Tea or Cocoa', 1901, 1),
 ('coffee, tea', 1903, 1),
 ('After Dinner Coffee Cup', 1905, 4),
 ('tea, coffee or milk', 1906, 5),
 ('Iced Tea of Coffee', 1907, 1),
 ('Demi Tasse de Mocha', 1908, 1),
 ('Coffeïnfreier Kaffee', 1910, 56),
 ('Décaféiné', 1911, 1),
 ('Le Cafe', 1912, 13),
 ('Buttered Toast and Coffee', 1913, 6),
 ('CAFE BRULOT', 1914, 5),
 ('Iced coffee, glass', 1915, 7),
 ('Mocha Cream Layer Cake', 1916, 4),
 ('Coffee, Tea, or Milk', 1917, 2),
 ('Coffee pot for one', 1918, 4),
 ('SANKA COFFEE', 1919, 220),
 ('Decaffeinated Coffee', 1920, 50),
 ('Mocha Cream Pie', 1921, 2),
 ('Sanka Coffee (Pot)', 1922, 12),
 ('Small Black Coffee', 1924, 4),
 ('Kaffee Hag, pot,', 1925, 1),
 ('Special Easter Dinner (pt.2): or Roast Prime Rib of Beef au Jus; Cauliflower in Butter, Sweet Potato Glace with Apple, Mixed Salad, Ice Cream with Crushed Fresh Strawberries and Cake, Demi-Tasse', 1926, 1),
 ('Doppio ristretto in tazza', 1927, 1),
 ('All Seafood Dinners include Soup, Salad, Vegetable, French Fried Potatoes, Rolls, Butter and Coffee', 1928, 1),
 ('Espresso or Cappuccino', 1929, 1),
 ('Sanka Coffee (Pot for One)', 1930, 9),
 ('Coffee (Hot or Iced)', 1931, 28),
 ('Arabica Coffee', 1932, 1),
 ('AMERICAN COFFEE', 1933, 141),
 ('Cafe Americain', 1934, 52),
 ('Idee Coffee', 1935, 8),
 ('Coffee plain', 1936, 1),
 ('Haagen-Dazs Coffee', 1937, 1),
 ('FRUHSTUCK mit Kaffee, Nescafe, Kaffee Hag, Tee, Kakao oder Milch und Brot, Brotchen, Zwieback, Hornchen, Knackebrot sowie Butter, Konfituren, Honig', 1938, 1),
 ('Cafe Decafeine', 1939, 9),
 ('Coffee Complet', 1940, 1),
 ('Espresso (35 cents extra)', 1941, 1),
 ('Coffee Liqueur', 1942, 1),
 ('Drip-O-Lator Sanka Coffee (Pot)', 1943, 2),
 ('Cafe creme', 1944, 8),
 ('Cup of Soup Omelette with Preserved Raspberries Florida New Potatoes in Cream Buttered Garden Peas Assorted Bread or Muffins Coffee Tea or Milk', 1945, 1),
 ('Le cafe viennois', 1946, 1),
 ("Medaglia d'Oro Coffee", 1947, 26),
 ('coffee HAG', 1948, 9),
 ('Coffee (Cup, Pot)', 1949, 1),
 ('French Ice Cream: Coffee', 1950, 7),
 ('cafe cona', 1951, 3),
 ('AMERICANO', 1952, 22),
 ('nescafe', 1953, 113),
 ('Gaelic Coffee', 1954, 5),
 ('Cafe Hag', 1955, 4),
 ('Caffe Hag', 1956, 2),
 ('Portion Filterkaffee', 1957, 2),
 ('Cappuccino', 1958, 73),
 ('Capuccino', 1959, 9),
 ('Cafe Maison', 1960, 0),
 ('Cafe Expresso', 1961, 6),
 ('Coffee or Tea per Cup', 1962, 1),
 ('Kona Coffee Grog', 1963, 2),
 ('Expresso', 1964, 15),
 ('Wiener Eiskaffee', 1965, 3),
 ('Cappucino, Chocolat', 1966, 1),
 ('Café o Té', 1967, 1),
 ('Sanka (non-Caffeine Coffee)', 1968, 2),
 ('Coffee Diablo for two (ten minutes)', 1969, 1),
 ('Eiscafe', 1970, 4),
 ('Instant Coffee (nescafe)', 1971, 1),
 ('Cafe expresso 100% Arabica', 1972, 1),
 ('Cafe (Petits Fours Secs)', 1973, 1),
 ('ordinary coffee', 1974, 1),
 ('Cafe decaf', 1975, 1),
 ('Les Cafes et Thes', 1976, 1),
 ('Coffee, tea, or chocolate.  Rolls, Croissant or Brioche. Marmalade, Butter.', 1977, 1),
 ('Irish Coffee A Blend  of Coffee and Irish Whiskey Topped with Whipped Cream', 1978, 1),
 ('Coffee Espresso', 1979, 2),
 ('Capuchino', 1980, 2),
 ('Mokka, Espresso klein', 1981, 1),
 ("Chemists' Club Special Blend of Coffee (pot)", 1982, 1),
 ('Entrecote "Cafe Paris", Pommes frites', 1983, 1),
 ('The Palace Blend Coffee, Decaffeinated

### Q2: Based on the dataset, has coffee always been on restaurants’ menus since the earliest record available?

In [ ]:
# find minimum first_appeared date in Dish dataset, compare to minimum first_appeared date in DishCleaned dataset - filtering for invalid dates

%%sql
select min(first_appeared) from Dish where first_appeared > 1000

 * sqlite://
Done.


[(1851.0,)]

In [ ]:
%%sql
select min(first_appeared_fixed) from DishCleaned

 * sqlite://
Done.


[(1856,)]

The minimum first_appeared value for a dish in the Dish dataset is 1851, in the DishCleaned dataset with only coffee dishes it is 1882

### Q3: How did coffee dish prices change over time?

In [ ]:
%%sql
select name, first_appeared_fixed, lowest_price, highest_price, times_appeared from DishCleaned group by first_appeared_fixed

 * sqlite://
Done.


[('Coffee Royal', 1856, 0.0, 0.14, 3),
 ('Mocha Coffee', 1858, 0.0, 0.0, 8),
 ('Hot Coffee', 1862, 0.1, 0.5, 37),
 ('CAFE', 1866, 0.0, 30.0, 885),
 ('French Coffee', 1881, 0.1, 15.0, 517),
 ('coffee, per cup', 1882, 0.05, 10.0, 124),
 ('Cafe noir', 1883, 0.0, 0.15, 591),
 ('Cup Coffee with 2 Corn Muffins', 1884, 0.05, 5.0, 2),
 ('Coffee Cream', 1885, 0.25, 0.3, 3),
 ('Coffi', 1886, 0.0, 0.0, 4),
 ('Cafe Parfait', 1887, 0.0, 35.0, 647),
 ('Kaffee', 1888, 0.0, 0.0, 868),
 ('Caffe', 1889, 0.0, 3.0, 46),
 ('Coffee de turin', 1890, 0.0, 0.0, 1),
 ('Cafe au Lait', 1891, 0.0, 1.35, 21),
 ('Demi-tasse', 1892, 0.0, 15.0, 878),
 ('Coffee Cake', 1893, 0.05, 5.0, 135),
 ('Iced coffee', 1894, 0.0, 25.0, 602),
 ('Turkish Coffee', 1895, 0.0, 25.0, 623),
 ('Mousse au cafe a la Netherland', 1896, 0.3, 0.3, 6),
 ('Mocha', 1897, 0.25, 0.25, 57),
 ('Coffee, Pot', 1898, 0.05, 1000.0, 173),
 ('Kaffee,', 1899, 0.0, 0.0, 5),
 ('Coffee', 1900, 0.0, 30.0, 8484),
 ('Coffee, Tea or Cocoa', 1901, 0.05, 0.05, 1),
 ('coffee, tea', 1903, 0.4, 0.4, 1),
 ('After Dinner Coffee Cup', 1905, 0.0, 0.0, 4),
 ('tea, coffee or milk', 1906, 0.0, 0.05, 5),
 ('Iced Tea of Coffee', 1907, 0.25, 0.25, 1),
 ('Demi Tasse de Mocha', 1908, 0.0, 0.0, 1),
 ('Coffeïnfreier Kaffee', 1910, 0.0, 0.0, 56),
 ('Décaféiné', 1911, 0.0, 0.0, 1),
 ('Le Cafe', 1912, 0.0, 0.0, 13),
 ('Buttered Toast and Coffee', 1913, 0.1, 10.0, 6),
 ('CAFE BRULOT', 1914, 1.25, 1.5, 5),
 ('Iced coffee, glass', 1915, 0.1, 0.15, 7),
 ('Mocha Cream Layer Cake', 1916, 0.1, 0.9, 4),
 ('Coffee, Tea, or Milk', 1917, 0.05, 0.05, 2),
 ('Coffee pot for one', 1918, 0.1, 0.25, 4),
 ('SANKA COFFEE', 1919, 0.0, 1.45, 220),
 ('Decaffeinated Coffee', 1920, 0.0, 2.0, 50),
 ('Mocha Cream Pie', 1921, 0.15, 0.15, 2),
 ('Sanka Coffee (Pot)', 1922, 0.0, 0.0, 12),
 ('Small Black Coffee', 1924, 0.1, 0.15, 4),
 ('Kaffee Hag, pot,', 1925, 0.25, 0.25, 1),
 ('Special Easter Dinner (pt.2): or Roast Prime Rib of Beef au Jus; Cauliflower in Butter, Sweet Potato Glace with Apple, Mixed Salad, Ice Cream with Crushed Fresh Strawberries and Cake, Demi-Tasse', 1926, 0.0, 0.0, 1),
 ('Doppio ristretto in tazza', 1927, 0.0, 0.0, 1),
 ('All Seafood Dinners include Soup, Salad, Vegetable, French Fried Potatoes, Rolls, Butter and Coffee', 1928, 0.0, 0.0, 1),
 ('Espresso or Cappuccino', 1929, 0.0, 0.0, 1),
 ('Sanka Coffee (Pot for One)', 1930, 0.25, 0.35, 9),
 ('Coffee (Hot or Iced)', 1931, 0.45, 0.45, 28),
 ('Arabica Coffee', 1932, 0.0, 0.0, 1),
 ('AMERICAN COFFEE', 1933, 0.0, 1.45, 141),
 ('Cafe Americain', 1934, 1.25, 1.25, 52),
 ('Idee Coffee', 1935, 0.0, 0.0, 8),
 ('Coffee plain', 1936, 0.0, 0.0, 1),
 ('Haagen-Dazs Coffee', 1937, 0.55, 0.55, 1),
 ('FRUHSTUCK mit Kaffee, Nescafe, Kaffee Hag, Tee, Kakao oder Milch und Brot, Brotchen, Zwieback, Hornchen, Knackebrot sowie Butter, Konfituren, Honig', 1938, 0.0, 0.0, 1),
 ('Cafe Decafeine', 1939, 0.0, 0.0, 9),
 ('Coffee Complet', 1940, 0.0, 0.0, 1),
 ('Espresso (35 cents extra)', 1941, 0.35, 0.35, 1),
 ('Coffee Liqueur', 1942, 0.0, 0.0, 1),
 ('Drip-O-Lator Sanka Coffee (Pot)', 1943, 0.15, 0.2, 2),
 ('Cafe creme', 1944, 0.0, 0.6, 8),
 ('Cup of Soup Omelette with Preserved Raspberries Florida New Potatoes in Cream Buttered Garden Peas Assorted Bread or Muffins Coffee Tea or Milk', 1945, 0.0, 0.0, 1),
 ('Le cafe viennois', 1946, 0.0, 0.0, 1),
 ("Medaglia d'Oro Coffee", 1947, 0.0, 0.0, 26),
 ('coffee HAG', 1948, 0.0, 0.0, 9),
 ('Coffee (Cup, Pot)', 1949, 0.1, 0.2, 1),
 ('French Ice Cream: Coffee', 1950, 0.6, 1.5, 7),
 ('cafe cona', 1951, 0.0, 0.0, 3),
 ('AMERICANO', 1952, 0.0, 5.0, 22),
 ('nescafe', 1953, 0.0, 12.0, 113),
 ('Gaelic Coffee', 1954, 0.0, 0.0, 5),
 ('Cafe Hag', 1955, 0.0, 0.0, 4),
 ('Caffe Hag', 1956, 0.0, 0.0, 2),
 ('Portion Filterkaffee', 1957, 0.0, 0.0, 2),
 ('Cappuccino', 1958, 0.0, 4.5, 73),
 ('Capuccino', 1959, 0.0, 3.0, 9),
 ('Cafe Maison', 1960, 0.0, 0.0, 0),
 ('Cafe Expresso', 1961, 0.0, 2.0, 6),
 ('Coffee or Tea per Cup', 1962, 0.0, 0.0, 1),
 ('Kona Coffee Grog', 1963, 0.0, 0.0, 